pkg/kor/formatter.go

```diff
@@ -0,0 +1,209 @@
+		if err := json.Unmarshal(jsonResponse, &resources); err != nil {
+			return "", err
+		}
+
+		if !opts.ShowReason && outputFormat == "json" {
+			// Create a map of namespaces with their corresponding maps of resource types and lists of resource names
+			namespaces := make(map[string]map[string][]string)
+			for namespace, resourceMap := range resources {
+				for resourceType, infoSlice := range resourceMap {
+					for _, info := range infoSlice {
+						if _, ok := namespaces[namespace]; !ok {
+							namespaces[namespace] = make(map[string][]string)
+						}
+						namespaces[namespace][resourceType] = append(namespaces[namespace][resourceType], info.Name)
+					}
+				}
+			}
+			// Marshal the map to JSON format
+			modifiedJSONResponse, err := json.MarshalIndent(namespaces, "", "  ")
+			if err != nil {
+				return "", err
+			}
+			return string(modifiedJSONResponse), nil
+		}
+
+		// Marshal JSON response with reasons
+		modifiedJSONResponse, err := json.MarshalIndent(resources, "", "  ")
+		if err != nil {
+			return "", err
+		}
+		return string(modifiedJSONResponse), nil
+	default:
+		return "", fmt.Errorf("unsupported output format: %s", outputFormat)
+	}
+}
+
+func FormatOutput(resources map[string]map[string][]ResourceInfo, opts Opts) bytes.Buffer {
+	var output bytes.Buffer
+	switch opts.GroupBy {
+	case "namespace":
+		for namespace, diffs := range resources {
+			output.WriteString(formatOutputForNamespace(namespace, diffs, opts))
+		}
+	case "resource":
+		for resource, diffs := range resources {
+			output.WriteString(formatOutputForResource(resource, diffs, opts))
+		}
+	}
+	return output
+}
+
+func formatOutputForNamespace(namespace string, resources map[string][]ResourceInfo, opts Opts) string {
+	var buf strings.Builder
+	table := tablewriter.NewWriter(&buf)
+	table.SetColWidth(60)
+	table.SetHeader(getTableHeader(opts.GroupBy, opts.ShowReason))
+	allEmpty := true
+	var index int
+	for resourceType, diff := range resources {
+		for _, info := range diff {
+			row := getTableRow(index, resourceType, info.Name)
+			if opts.ShowReason && info.Reason != "" {
+				row = append(row, info.Reason)
+			}
+			table.Append(row)
+			allEmpty = false
+			index++
+		}
+	}
+	if allEmpty {
+		if opts.Verbose {
+			return fmt.Sprintf("No unused resources found in the namespace: %q\n", namespace)
+		}
+		return ""
+	}
+
+	table.Render()
+	return fmt.Sprintf("Unused resources in namespace: %q\n%s\n", namespace, buf.String())
+}
+
+func formatOutputForResource(resource string, resources map[string][]ResourceInfo, opts Opts) string {
+	if len(resources) == 0 {
+		if opts.Verbose {
+			return fmt.Sprintf("No unused %ss found\n", resource)
+		}
+		return ""
+	}
+	var buf bytes.Buffer
+	table := tablewriter.NewWriter(&buf)
+	table.SetColWidth(20)
+	table.SetHeader(getTableHeader(opts.GroupBy, opts.ShowReason))
+	var index int
+	for _, infos := range resources {
+		for _, info := range infos {
+			row := []string{info.Name}
```

Comment:
Outputing by resource breaks the table.

```suggestion
	table.SetColWidth(60)
	table.SetHeader(getTableHeader(opts.GroupBy, opts.ShowReason))
	var index int
	for ns, infos := range resources {
		for _, info := range infos {
			row := getTableRow(index, ns, info.Name)
```

Reply:
Before fix:
![image](https://github.com/yonahd/kor/assets/70585638/9017b540-9196-427b-81c6-ec826f4155f4)

After fix:
![image](https://github.com/yonahd/kor/assets/70585638/07a8ac53-def7-4eb2-a7bd-d1f595a6f1cd)